crates/astria-config-attribute/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use itertools::Itertools;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse_macro_input,
+    spanned::Spanned,
+    Ident,
+    Item,
+};
+
+fn extract_struct_properties(item_ast: &Item) -> syn::Result<syn::Ident> {
+    let item_top_level_span = item_ast.span();
+    let Item::Struct(ref item_struct) = item_ast else {
+        return Err(syn::Error::new(
+            item_top_level_span,
+            "the astria_config proc macro can only be called on structs",
+        ));
+    };
+
+    let struct_name = item_struct.ident.clone();
+
+    // Generates an vec of token-tree like structures of the atomicity we care about
+    let struct_fields: Vec<Ident> = item_struct
+        .fields
+        .iter()
+        .map(|x| {
+            x.ident.clone().ok_or(syn::Error::new(
+                item_top_level_span,
+                "Missing field identifier in struct",
+            ))
+        })
+        .try_collect()?;
+
+    // Check if log is a valid member of the config, otherwise throw compile error
+    let true = struct_fields
```

Comment:
replace this by `if struct_fields...contains()` instead of having this odd `let pat = ... else` construct.